### PR TITLE
Fix bug 780152 - Fixed css for wiki edit image

### PR DESCRIPTION
**Pull request by Moemonty**
Fixed css for wiki image editor.
image-editor-fixed

### CODE REVIEW

**Comment by groovecoder** [comment]
Thanks @Moemonty! Sorry to be the fun police, but [we'll need to re-word commit message so each one starts with "bug NNNNNNN" or "fix bug NNNNNNN"](https://github.com/mozilla/kuma/blob/master/CONTRIBUTING.md#contributing-code).
giphy

We can help do that for you if you want.

**Comment by Moemonty** [comment]
I saw that. That's why I recommitted here.
https://github.com/Moemonty/kuma/commit/08d1bbeddb6368664348d59a547678a49b09906b


On Wed, Jan 7, 2015 at 11:28 AM, luke crouch <notifications@github.com>
wrote:

> Thanks @Moemonty <https://github.com/Moemonty>! Sorry to be the fun
> police, but we'll need to re-word commit message so each one starts with
> "bug NNNNNNN" or "fix bug NNNNNNN"
> <https://github.com/mozilla/kuma/blob/master/CONTRIBUTING.md#contributing-code>
> .
> [image: giphy]
> <https://cloud.githubusercontent.com/assets/71928/5649967/06676096-9660-11e4-82c1-493169ea3779.gif>
>
> We can help do that for you if you want.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/kuma/pull/2986#issuecomment-69057621>.
>



-- 


*Moises MontenegroMobile: 847-894-0481Web: http://www.moisesmontenegro.com
<http://www.moisesmontenegro.com>*

**Comment by groovecoder** [comment]
Hmm ... it's here on GitHub as 2 separate commits: https://github.com/Moemonty/kuma/commit/107c99e0445693020281aa40da8c13268cc7d647 and https://github.com/Moemonty/kuma/commit/08d1bbeddb6368664348d59a547678a49b09906b. Can we squash these into a single commit?

**Comment by Moemonty** [comment]
I believe these are merged now.

**Comment by groovecoder** [comment]
https://github.com/Moemonty/kuma/commit/107c99e0445693020281aa40da8c13268cc7d647 is still included on its own. If you like, I can take a stab at squashing them and re-submitting a PR for you?

**Comment by Moemonty** [comment]
Sure! I could just start a new branch or try again. Just let me know!

Thanks,
Moises M.



On Wed, Jan 7, 2015 at 3:02 PM, luke crouch <notifications@github.com>
wrote:

> Moemonty@107c99e
> <https://github.com/Moemonty/kuma/commit/107c99e0445693020281aa40da8c13268cc7d647>
> is still included on its own. If you like, I can take a stab at squashing
> them and re-submitting a PR for you?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/kuma/pull/2986#issuecomment-69091444>.
>



-- 


*Moises MontenegroMobile: 847-894-0481Web: http://www.moisesmontenegro.com
<http://www.moisesmontenegro.com>*

**Comment by openjck** [comment]
There are a few different ways we could correct the commits, but the following steps will be easiest:

1. Run `git checkout master` to check out the default branch
2. Run `git branch -D new-issue-780152` to delete the branch you created earlier
3. Run `git checkout -b new-issue-780152` to create and switch to a new branch with the same name
4. Make <a href="https://github.com/mozilla/kuma/pull/2986/files#diff-52032a07d2ab5ddd66541f81c351ef51R107">the changes</a> again
5. Commit the changes. Make sure the commit message starts with "Fix bug 780152".
6. Run `git push -fu origin new-issue-780152` to send your new commit to this pull request

**Comment by Moemonty** [comment]
Ok, one commit with one change. Here we go!

**Comment by openjck** [comment]
Thanks so much for correcting the commit. Looks much better now.

I still see this bug after applying the changes. Maybe we can do some digging together...

---

The bottom part of the image attachment pop-up is broken up into a left side and a right side. A table cell is used to represent each.

Left side:

```html
<td class="cke_dialog_ui_hbox_first" ... >
```

Right side:

```html
<td class="cke_dialog_ui_hbox_last" ... >
```

According to Firefox Developer Tools, each side is supposed to have a 50% width. But if we look closely, we can see that the right side is much bigger than it should be.

Let's take a look at the preview box...

```html
<div class="ImagePreviewBox">
```

It has a fixed width of `300px`. That makes the right side grow to be bigger than it should. If we set the width of the preview box to be `100%` instead, it will be exactly as big as the right side is supposed to be.

Now each side has a `50%` width as intended. That's a step in the right direction, but the problem still exists.

Let's look at the input containers...

```html
<div class="cke_dialog_ui_input_text" ... >
```

They both have fixed widths of `40px`, but that's too small to fit the input boxes. If we set the widths to `100%` instead, they grow to be as big as possible, which ends up being half of the left side. That does the trick.

So to summarize, we can correct the bug with two changes

* Set the width of the image preview box (`.ImagePreviewBox`) to be `100%`.
* Set the width of the input containers (`.cke_dialog_ui_input_text`) to be `100%`.

We only want the changes to affect <em>this</em> pop-up window, so we should make sure to match against `#cke_dialog_contents_78` as well.

The resulting CSS might look something like this, but it may be slightly different. I'll leave the final testing and implementation up to you.

```css
#cke_dialog_contents_78 .ImagePreviewBox, #cke_dialog_contents_78 .cke_dialog_ui_input_text {
    width: 100%;
}
```

I hope that helps. Let me know if there's anything else I can do to help.

**Comment by Moemonty** [comment]
https://www.dropbox.com/lightbox/home/Screenshots?select=Screenshot+2015-01-11+18.53.31.png
Screenshot attached.

**Comment by Moemonty** [comment]
screenshot 2015-01-11 18 53 31

**Comment by openjck** [comment]
Awesome! Thanks for making the updates. There's one last thing I'll add a comment about. Aside from that this looks great!

File: media/js/libs/ckeditor/skins/kuma/dialog.css [hunk 1]
```diff
@@ -104,3 +104,21 @@ body .cke_dialog{visibility:visible;padding:20px;background:rgba(0,0,0,.3);font:
...
+
+.cke_skin_kuma #cke_dialog_contents_78 div.cke_dialog_ui_input_text, 
+.cke_skin_kuma #cke_dialog_contents_78 div.cke_dialog_ui_input_password {
+  border:none;
+}
+
+.cke_skin_kuma #cke_dialog_contents_78 input.cke_dialog_ui_input_text, 
+.cke_skin_kuma #cke_dialog_contents_78 input.cke_dialog_ui_input_password {
+  border:solid 1px #cbc8b9;
+}
```
Comment (openjck) [inline media/js/libs/ckeditor/skins/kuma/dialog.css]: I notice that these changes cause the input borders to appear mis-aligned with the input headings. In other words, the left side of the border appears a few pixels off from the left side of the heading.

It looks like the bug is still corrected without these changes. Was there another reason these changes were added? If not, we could safely remove them.

Comment (Moemonty) [inline media/js/libs/ckeditor/skins/kuma/dialog.css]: The borders appeared off and resulted in the original visual anomaly. I'll look again!

**Comment by openjck** [comment]
It looks like our recent upgrade to CKEditor 4 fixed the same bug. I'm sorry for not considering that possibility sooner. :frowning:

You did a great job on this pull request. We have a [list of other good first bugs](https://bugzilla.mozilla.org/buglist.cgi?j_top=OR&list_id=11858643&o1=isnotempty&o2=substring&v2=%5Bgood%20first%20bug%5D&f1=bug_mentor&columnlist=short_desc%2Ccomponent%2Cchangeddate&query_format=advanced&f2=status_whiteboard&bug_status=UNCONFIRMED&bug_status=NEW&bug_status=ASSIGNED&bug_status=REOPENED&product=Mozilla%20Developer%20Network) if you're still interested in helping out.

**Comment by Moemonty** [comment]
Absolutely. I'll find a bug on the list. Thanks!

**Comment by Moemonty** [comment]
Hey Johh,

I'd like to try this one.
https://bugzilla.mozilla.org/show_bug.cgi?id=1118723

On Fri, Jan 16, 2015 at 12:58 PM, John Karahalis <notifications@github.com>
wrote:

> It looks like our recent upgrade to CKEditor 4 fixed the same bug. I'm
> sorry for not considering that possibility sooner. [image: :frowning:]
>
> You did a great job on this pull request. We have a list of other good
> first bugs
> <https://bugzilla.mozilla.org/buglist.cgi?j_top=OR&list_id=11858643&o1=isnotempty&o2=substring&v2=%5Bgood%20first%20bug%5D&f1=bug_mentor&columnlist=short_desc%2Ccomponent%2Cchangeddate&query_format=advanced&f2=status_whiteboard&bug_status=UNCONFIRMED&bug_status=NEW&bug_status=ASSIGNED&bug_status=REOPENED&product=Mozilla%20Developer%20Network>
> if you're still interested in helping out.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/kuma/pull/2986#issuecomment-70303513>.
>



-- 


*Moises MontenegroMobile: 847-894-0481Web: http://www.moisesmontenegro.com
<http://www.moisesmontenegro.com>*

**Comment by openjck** [comment]
[bcc: GitHub]

Great choice. Thanks for your help on this--it's an important addition.

I'll be available via email and IRC (openjck in #mdndev) if there's
anything I can to do help.

On Mon, Jan 26, 2015 at 10:31 AM, Moises Montenegro <
notifications@github.com> wrote:

> Hey Johh,
>
> I'd like to try this one.
> https://bugzilla.mozilla.org/show_bug.cgi?id=1118723
>
> On Fri, Jan 16, 2015 at 12:58 PM, John Karahalis <notifications@github.com>
>
> wrote:
>
> > It looks like our recent upgrade to CKEditor 4 fixed the same bug. I'm
> > sorry for not considering that possibility sooner. [image: :frowning:]
> >
> > You did a great job on this pull request. We have a list of other good
> > first bugs
> > <
> https://bugzilla.mozilla.org/buglist.cgi?j_top=OR&list_id=11858643&o1=isnotempty&o2=substring&v2=%5Bgood%20first%20bug%5D&f1=bug_mentor&columnlist=short_desc%2Ccomponent%2Cchangeddate&query_format=advanced&f2=status_whiteboard&bug_status=UNCONFIRMED&bug_status=NEW&bug_status=ASSIGNED&bug_status=REOPENED&product=Mozilla%20Developer%20Network>
>
> > if you're still interested in helping out.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/mozilla/kuma/pull/2986#issuecomment-70303513>.
> >
>
>
>
> --
>
>
> *Moises MontenegroMobile: 847-894-0481Web: http://www.moisesmontenegro.com
> <http://www.moisesmontenegro.com>*
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/kuma/pull/2986#issuecomment-71478388>.
>



-- 
John Karahalis
Mozilla
openjck.com
